src/node/chainstate.cpp

```diff
@@ -207,7 +207,9 @@ ChainstateLoadResult LoadChainstate(ChainstateManager& chainman, const CacheSize
     } else if (snapshot_completion == SnapshotCompletionResult::SUCCESS) {
         LogPrintf("[snapshot] cleaning up unneeded background chainstate, then reinitializing\n");
         if (!chainman.ValidatedSnapshotCleanup()) {
-            AbortNode("Background chainstate cleanup failed unexpectedly.");
+            const auto error = Untranslated("Background chainstate cleanup failed unexpectedly.");
+            AbortNode(error.original);
+            return {ChainstateLoadStatus::FAILURE, error};
```

Comment:
Just an observation on the behavior change here. In the current behavior, the function continues execution until it reaches `CompleteChainstateInitialization` further below. There it calls `options.check_interrupt()`, which will lead it to return `ChainstateLoadStatus::INTERRUPTED`. I think making it return a `ChainstateLoadStatus::FAILURE` now makes more sense, since it is not stopped due to an external interrupt.